test/src/e2e_vm_tests/test_programs/should_fail/impl_self_duplicated/test.toml

```diff
@@ -0,0 +1,6 @@
+category = "fail"
+validate_abi = false
+expected_warnings = 3
+
+# check: fn foo(self) -> u64 {
+# nextln: $()Duplicate definitions for the method "foo" for type "S<T>".
```

Comment:
This means that the error is triggered when `impl<T> S<T> where T : Dog { ... }` is typechecked.

As far as I can tell Rust doesn't throw the error until `s.foo()` is typechecked in line 41 of the test program.

I'm happy with the current behavior, but it's just worth being aware that we are diverging from Rust here.